apps/dh/api-dh/source/DataHub.WebApi.Tests/Unit/Clients/Wholesale/SettlementReports/SettlementReportsClientTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Energinet.DataHub.WebApi.Clients.Wholesale.SettlementReports;
+using Energinet.DataHub.WebApi.Clients.Wholesale.SettlementReports.Dto;
+using Energinet.DataHub.WebApi.Clients.Wholesale.v3;
+using Moq;
+using Moq.Protected;
+using NodaTime;
+using NodaTime.Text;
+using Xunit;
+
+namespace Energinet.DataHub.WebApi.Tests.Unit.Clients.Wholesale.SettlementReports;
+
+public class SettlementReportsClientTests : IDisposable
+{
+    private readonly Mock<HttpMessageHandler> _httpMessageHandlerMock;
+    private readonly HttpClient _httpClient;
+    private readonly SettlementReportsClient _client;
+
+    public SettlementReportsClientTests()
+    {
+        _httpMessageHandlerMock = new Mock<HttpMessageHandler>();
+        _httpClient = new HttpClient(_httpMessageHandlerMock.Object);
+        _httpClient.BaseAddress = new Uri("http://localhost/");
+        _client = new SettlementReportsClient("http://localhost", _httpClient, _httpClient, _httpClient);
+    }
+
+    [Theory]
+    [InlineData("2024-11-01T23:00:00.000Z", "2024-12-31T22:59:59.999Z")]
```

Comment:
Doesn't the frontend send `2024-12-31T23:00:00Z` as the to date, if not, the `AddMilliseconds(-1)` does not make sense?